### PR TITLE
Docs for @ember/component

### PR DESCRIPTION
**Pull request by NullVoxPopuli**

### CODE REVIEW

File: packages/@ember/component/index.ts [hunk 1]
```diff
@@ -1,6 +1,141 @@
...
+ *   capabilities = capabilities('3.13', {
+ *     // capabilities listed here
+ *   })
+ * }
+ * ```
+ *
+ *
+ * For a full list of capabilities, their defaults, and how they are used, see [@glimmer/manager](https://github.com/glimmerjs/glimmer-vm/blob/4f1bef0d9a8a3c3ebd934c5b6e09de4c5f6e4468/packages/%40glimmer/manager/lib/public/component.ts#L26)
+ *
+ * @param {'3.4' | '3.13'} managerApiVersion
```
Comment (chriskrycho) [inline packages/@ember/component/index.ts:124]: I *believe* this has to be `3.13` for Ember v4?

**Review by chriskrycho — COMMENTED** [review]
Thanks very much for opening this!

**Comment by wagenet** [comment]
Might be some overlap with #19948.

**Comment by chriskrycho** [comment]
@NullVoxPopuli mind rebasing this and addressing the comment above? (The rebase doesn’t look strictly necessary but given the amount of change it’d make me more confident in the results here!)

**Comment by NullVoxPopuli** [comment]
@chriskrycho done. thanks!

File: packages/@ember/component/index.ts [hunk 2]
```diff
@@ -1,6 +1,141 @@
...
+ *   // ...
+ * }
+ *
+ * let theTemplateFactory = getTemplateFactory(Demo)
+ * ```
+ *
+ * @param {object} componentDefinition
+ * @returns {TemplateFactory}
+ */
+export const getComponentTemplate = _getComponentTemplate;
```
Comment (chriskrycho) [inline packages/@ember/component/index.ts:58]: As above!

**Review by chriskrycho — CHANGES_REQUESTED** [review]

**Comment by chriskrycho** [comment]
@NullVoxPopuli still interested in landing these docs updates?

**Comment by NullVoxPopuli** [comment]
@chriskrycho updated!

**Comment by locks** [comment]
@NullVoxPopuli is this still relevant?

**Comment by NullVoxPopuli** [comment]
Yup

**Comment by NullVoxPopuli** [comment]
Ember helper is also missing a bunch of docs

**Comment by kategengler** [comment]
@NullVoxPopuli If this is still relevant, do you want to update it?

**Comment by NullVoxPopuli** [comment]
Yes, this documentation has been missing for a long time :upside_down_face: update incoming

**Comment by kategengler** [comment]
Enabled auto-merge, thank you!

**Pull request merged**